python/graphstorm/gconstruct/transform.py

```diff
@@ -1725,76 +1725,100 @@ def parse_label_ops(confs, is_node):
-                "'label_col' must be defined in the label field."
-        label_col = label_conf['label_col']
-        return [ClassificationProcessor(label_col, label_col, split_pct,
-                                        label_stats_type, mask_field_names)]
-    elif task_type == 'regression':
-        assert 'label_col' in label_conf, \
-                "'label_col' must be defined in the label field."
-        label_col = label_conf['label_col']
-        return [RegressionProcessor(label_col, label_col, split_pct,
-                                    label_stats_type, mask_field_names)]
-    else:
-        assert task_type == 'link_prediction', \
-                "The task type must be classification, regression or link_prediction."
-        assert not is_node, "link_prediction task must be defined on edges."
-        return [LinkPredictionProcessor(None, None, split_pct,
-                                        label_stats_type, mask_field_names)]
+        if 'split_pct' in label_conf:
+            split_pct = label_conf['split_pct']
+        else:
+            logging.info("'split_pct' is not found. " + \
+                    "Use the default data split: train(80%), valid(10%), test(10%).")
+            split_pct = [0.8, 0.1, 0.1]
+
+        if task_type == 'classification':
+            assert 'label_col' in label_conf, \
+                    "'label_col' must be defined in the label field."
+            label_col = label_conf['label_col']
+            return ClassificationProcessor(label_col, label_col, split_pct,
+                                            label_stats_type, mask_field_names)
+        elif task_type == 'regression':
+            assert 'label_col' in label_conf, \
+                    "'label_col' must be defined in the label field."
+            label_col = label_conf['label_col']
+            return RegressionProcessor(label_col, label_col, split_pct,
+                                        label_stats_type, mask_field_names)
+        else:
+            assert task_type == 'link_prediction', \
+                    "The task type must be classification, regression or link_prediction."
+            assert not is_node, "link_prediction task must be defined on edges."
+            return LinkPredictionProcessor(None, None, split_pct,
+                                            label_stats_type, mask_field_names)
+    label_ops = []
+    for label_conf in label_confs:
+        label_ops.append(parse_label_conf(label_conf))
+
+    if len(label_ops) > 1:
+        # check whether train/val/test mask names are
+        # different for different labels
+        mask_names = []
+        for ops in label_ops:
+            mask_names.append(ops.train_mask_name)
+            mask_names.append(ops.val_mask_name)
+            mask_names.append(ops.test_mask_name)
+        assert len(mask_names) == len(set(mask_names)), \
+            f"Some train/val/test mask field names are duplicated, please check: {mask_names}."
```

Comment:
I think the code here will check all the masks field name for both node and edge. We should include that in the doc about that each mask name for different types of nodes/edges should be different.

Reply:
mask names for different node types can be the same. This function is called per node/edge type.